bokeh/models/widgets/widget.py

```diff
@@ -17,3 +17,15 @@ class Widget(LayoutDOM):
     ''' A base class for all interactive widget types.
 
     '''
+
+class NeedsJQuery(object):
+    """ A mixin for widgets requiring JQuery and/or JQuery UI. """
+
+    __javascript__ = [
+        "https://code.jquery.com/jquery-3.2.1.min.js",
+        "https://code.jquery.com/ui/1.12.1/jquery-ui.min.js",
+    ]
+
+    __css__ = [
+        "https://code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css",
+    ]
```

Comment:
> : A very important use case that must be considered is people working on isolated networks (...)

I didn't consider this yet, as this touch a much broader topic that wasn't addressed when external resources feature was added. Another situation to consider is people not wanting this being included automatically at all. I'm going to address those in future PRs in this release cycle.